src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -267,12 +268,13 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
                                @Nullable String kmsKeyName, BigQueryFileFormat sourceFormat, String writeDisposition,
                                List<String> gcsPaths, PartitionType partitionType, @Nullable Range range,
                                @Nullable String partitionByField, boolean requirePartitionFilter,
-                               List<String> clusteringOrderList, boolean tableExists, String jobId)
-      throws IOException, InterruptedException {
+                               List<String> clusteringOrderList, boolean tableExists, Configuration conf)
+      throws Exception {
```

Comment:
Why are we throwing the generic exception instead of the specific ones?

Reply:
Because one of the old method was throwing "Exception". On including that in this method automatically made all other Exceptions redundant. I did not touch that old method. 
Also The parent method calling this just throws IOException in try catch, irrespective of the type of error. Check line 228 .

Reply:
Refactored a bit and reverted the exceptions.